### PR TITLE
Fixes #303. Replaced dynamic styling with class-based CSS approach.

### LINKED ISSUE #303: [Feature] Replace the dynamic styling in listing.js with class-based CSS approach
## Problem
In the `_assets/js/listing.js` file, to make the resources visible, the use of dynamic styling has been done.

## Description
Instead of creating a new style element for the displaying of resource cards, It will be better to use a css property of `display: none` via a class based approach. we can create a new utility class for the task.
The link to `listing.js` file: https://github.com/creativecommons/cc-resource-archive/blob/main/docs/_assets/js/listing.js

## Implementation

- [x] I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by poojatalele**
## Fixes

- Fixes #303 by @Murdock9803 

## Description

This PR addresses the issue where dynamic styling was being used to control the visibility of resource cards in the _assets/js/listing.js file. To enhance maintainability and consistency, the following updates were made:

Introduced new CSS utility classes (.hidden and .visible) to handle the visibility of resource elements without relying on dynamically injected styles.
Refactored the updateVisibility function to utilize these new classes, ensuring that elements are shown or hidden based on the user's filter selections (topic, medium, language).
Adjusted the resource navigation filters to dynamically apply the appropriate visibility classes, providing a cleaner and more efficient way to control the display of filters.
These changes align with the suggestion to replace dynamic styling with a class-based approach, making the codebase more consistent and easier to maintain.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the *default* branch of the repository (`main` or `master`).
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [ ] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

For the purposes of this DCO, "license" is equivalent to "license or public domain dedication," and "open source license" is equivalent to "open content license or public domain dedication."
<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

**Comment by Murdock9803** [comment]
@poojatalele  you should make more descriptive Pull Requests. The fields are almost all empty. You can remove them if not necessary.

File: docs/Gemfile.lock [hunk 1]
```diff
@@ -270,6 +270,7 @@ GEM
 PLATFORMS
   aarch64-linux
   arm64-darwin-23
+  x86_64-linux
```
Comment (Murdock9803) [inline docs/Gemfile.lock:273]: This is an additional change which should be reverted. The `Gemfile.lock` file is not the target here i believe.

**Comment by poojatalele** [comment]
> @poojatalele you should make more descriptive Pull Requests. The fields are almost all empty. You can remove them if not necessary.

Updated all the changes, kindly review.

**Comment by possumbilities** [comment]
@poojatalele thank you for this contribution!

After looking through there are some improvements that could be made to streamline it a bit further. Since the default state is visible, we don't need a `visible` class, and could instead only have a `hidden` class and utilize the `classList.toggle` method rather than the `classList.add` and `classList.remove` methods.

For a somewhat similar example of approach see the `vocabulary.js` file.

**Comment by poojatalele** [comment]
> @poojatalele thank you for this contribution!
> 
> After looking through there are some improvements that could be made to streamline it a bit further. Since the default state is visible, we don't need a `visible` class, and could instead only have a `hidden` class and utilize the `classList.toggle` method rather than the `classList.add` and `classList.remove` methods.
> 
> For a somewhat similar example of approach see the `vocabulary.js` file.

Okay, got it! I'll update the changes.

**Comment by possumbilities** [comment]
@poojatalele Thank you! 👍

**Comment by poojatalele** [comment]
@Murdock9803 @possumbilities Replaced ```classList.add``` and ```classList.remove``` with ```classList.toggle```.
Kindly review :)

**Comment by possumbilities** [comment]
The changes here break the function of the filters after recent UX work. Closing, and would suggest it be revisitied with current work merged in.

**Pull request closed without merging**